EventsExpress/ClientApp/src/components/event/layouts/carousel/change-visit-status-button.js

```diff
@@ -0,0 +1,85 @@
+﻿import React from 'react';
+import moment from 'moment';
+import eventStatusEnum from '../../../../constants/eventStatusEnum';
+import Tooltip from '@material-ui/core/Tooltip';
+import IconButton from '@material-ui/core/IconButton';
+import { connect } from 'react-redux';
+import { join, leave } from '../../../../actions/event/event-item-view-action';
+import EventLeaveModal from '../../event-leave-modal';
+import { useFilterActions } from '../../filter/filter-hooks';
+import { get_events } from '../../../../actions/event/event-list-action';
+
+const ChangeVisitStatusButton = ({ event, currentUser, ...props }) => {
+    const { id, dateFrom, members, organizers, maxParticipants, isOnlyForAdults, eventStatus } = event;
+    const { getQueryWithRequestFilters } = useFilterActions();
+
+    const refreshEvents = () => {
+        const filterQuery = getQueryWithRequestFilters();
+        props.getEvents(filterQuery);
+    };
+
+    const visitorsEnum = {
+        approvedUsers: members.filter(x => x.userStatusEvent === 0),
+        deniedUsers: members.filter(x => x.userStatusEvent === 1),
+        pendingUsers: members.filter(x => x.userStatusEvent === 2)
+    };
+
+    const today = moment().startOf('day');
```

Comment:
There are is just too much logic here. Should we extract into two different buttons and then implement some chain of responsibility pattern?

Reply:
Done